### PR TITLE
br/lightning: PdController uses PD HTTP client

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #49865

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
rely on https://github.com/pingcap/tidb/pull/50718

File: br/pkg/pdutil/pd_serial_test.go [hunk 1]
```diff
@@ -54,242 +62,49 @@ func TestScheduler(t *testing.T) {
...
-	v := []byte("\"v4.1.0-alpha1\"\n")
+	v := "\"v4.1.0-alpha1\"\n"
 	r := parseVersion(v)
 	expectV := semver.New("4.1.0-alpha1")
 	require.Equal(t, expectV.Major, r.Major)
 	require.Equal(t, expectV.Minor, r.Minor)
 	require.Equal(t, expectV.PreRelease, r.PreRelease)
 }
 
-func TestPDRequestRetry(t *testing.T) {
```
Comment (D3Hunter) [inline br/pkg/pdutil/pd_serial_test.go:172]: now pd http client will do retry internally?

File: br/pkg/pdutil/pd.go [hunk 2]
```diff
@@ -627,24 +378,15 @@ func (p *PdController) doUpdatePDScheduleConfig(
...
-			return errors.Trace(err)
-		}
-		_, e := post(ctx, addr, prefix,
-			p.cli, http.MethodPost, reqData)
-		if e == nil {
-			return nil
-		}
-		log.Warn("failed to update PD config, will try next", zap.Error(e), zap.String("pd", addr))
+	if err := p.pdHTTPCli.SetConfig(ctx, newCfg, ttlSeconds...); err != nil {
+		return errors.Annotate(berrors.ErrPDUpdateFailed, "failed to update PD schedule config")
```
Comment (D3Hunter) [inline br/pkg/pdutil/pd.go:382]: wrap `err` inside returned error?

File: br/pkg/pdutil/pd.go [hunk 3]
```diff
@@ -834,125 +576,47 @@ func (p *PdController) doRemoveSchedulersWith(
...
+	err := p.pdHTTPCli.ResetTS(ctx, ts, true)
+	if err == nil {
+		return nil
+	}
+	if strings.Contains(err.Error(), http.StatusText(http.StatusForbidden)) {
+		log.Info("reset-ts returns with status forbidden, ignore")
 		return nil
 	}
-	return errors.Trace(err)
+	return err
```
Comment (D3Hunter) [inline br/pkg/pdutil/pd.go:609]: seems some path of pd http client doesn't contains stack, so keep `errors.Trace(err)`?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by YuJuncen — APPROVED** [review]

**Comment by lance6716** [comment]
ping @D3Hunter

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**